frappe/permissions.py

```diff
@@ -171,10 +171,15 @@ def get_doc_permissions(doc, user=None, ptype=None):
 	def is_user_owner():
 		return (doc.get("owner") or "").lower() == user.lower()
 
-	if has_controller_permissions(doc, ptype, user=user) is False:
+	controller_permission = has_controller_permissions(doc, ptype, user=user)
+	
+ 	if controller_permission is False:
 		push_perm_check_log(_("Not allowed via controller permission check"))
 		return {ptype: 0}
 
+	if controller_permission is True:
+		return {ptype: 1}
```

Comment:
This is not how it's supposed to work. The current behaviour is correct. 

Controller hooks add additional validation, they cannot on their own grant permission which aren't present.

Reply:
@ankush The issue is when you have a custom permission logic in your custom app, how can you handle it if this method ignores the hooks permission method's result?